### PR TITLE
Use conda.txt instead of run.txt

### PR DESCRIPTION
**Pull request by bobleesj**

### CODE REVIEW

File: .github/workflows/_matrix-and-codecov-on-merge-to-main.yml [hunk 1]
```diff
@@ -56,8 +56,7 @@ jobs:
 
       - name: Install ${{ inputs.project }} and requirements
         run: |
-          conda install --file requirements/run.txt
-          conda install --file requirements/run-ext.txt
```
Comment (bobleesj) [inline .github/workflows/_matrix-and-codecov-on-merge-to-main.yml:60]: https://github.com/Billingegroup/release-scripts/pull/56#issuecomment-2386702356

as you've mentioned, I think adding `run-ext` adds an unwanted complexity. 

What if we both include optional and required dependencies all under `conda.txt` instead? 

We should test our software with all the dependencies installed anyway.. right?

Comment (bobleesj) [reply .github/workflows/_matrix-and-codecov-on-merge-to-main.yml:60]: Furthermore, if we decide to use have a separate .txt file for ext pacakges, then we also needed `pip-ext.txt` I don't think we want this..

**Review by bobleesj — COMMENTED** [review]
Ready for review! @sbillinge

**Comment by sbillinge** [comment]
yes, much better.  I like it.

**Pull request merged**